### PR TITLE
Refactor industry::update

### PR DESCRIPTION
**Pull request by IntelOrca**
* Fix industrymgr::update being called twice in a tick.
* Move some of the surface tile element operations to property methods.
* Extract code into smaller functions.

### CODE REVIEW

File: src/openloco/industrymgr.cpp [hunk 1]
```diff
@@ -38,8 +38,6 @@ namespace openloco::industrymgr
                 }
             }
         }
-
-        call(0x00453234);
```
Comment (AaronVanGeffen) [inline src/openloco/industrymgr.cpp:42]: This effectively meant industrymgr::update was being called twice in a tick, as you said. Does that mean industries were producing cargo twice as fast, too? This should probably be mentioned in the changelog.

Comment (IntelOrca) [reply src/openloco/industrymgr.cpp:42]: > Does that mean industries were producing cargo twice as fast, too?

I am not sure.

Comment (AaronVanGeffen) [reply src/openloco/industrymgr.cpp:42]: Tested it a bit; doesn't appear to have that sort of effect. There's RNG involved, though, so it's a little hard to test.

File: src/openloco/industry.cpp [hunk 2]
```diff
@@ -119,69 +119,82 @@ namespace openloco
...
-                        var_DF = 255;
-                    }
+            }
+        }
+    }
+
+    void industry::sub_453354()
+    {
+        auto obj = object();
+        int16_t tmp_a = var_DB / 16;
```
Comment (duncanspumpkin) [inline src/openloco/industry.cpp:162]: if we are refactoring might as well move this to `camelCase`

Comment (AaronVanGeffen) [reply src/openloco/industry.cpp:162]: Not necessary for this PR, I think.

File: src/openloco/industry.cpp [hunk 3]
```diff
@@ -119,69 +119,82 @@ namespace openloco
...
+            {
+                if (surface->industry_id() == id())
+                {
+                    uint8_t bl = surface->var_6_SLR5();
+                    auto obj = object();
+                    if (bl == 0 || bl != obj->var_EA)
                     {
-                        if (surface->industry_id() == id())
+                        var_DB++;
+                        if ((!(obj->flags & industry_object_flags::flag_28) && surface->var_4_E0() == 0) || find_5(surface))
```
Comment (marijnvdwerf) [inline src/openloco/industry.cpp:149]: I think this should be `surface->var_4_E0() != 0` (looking at `0x4532F8`)

Comment (AaronVanGeffen) [reply src/openloco/industry.cpp:149]: Yep, that seems to fix it. Could you file a PR?

**Pull request merged**